packages/api/lib/granule-remove-from-cmr.js

```diff
@@ -42,73 +40,50 @@ const _removeGranuleFromCmr = async (granule) => {
  */
 const unpublishGranule = async (
   knex,
-  granule,
+  pgGranuleRecord,
   granulePgModel = new GranulePgModel(),
   granuleDynamoModel = new models.Granule()
 ) => {
-  const collectionPgModel = new CollectionPgModel();
-  let pgGranuleCumulusId;
-
   // If we cannot find a Postgres Collection or Postgres Granule,
   // don't update the Postgres Granule, continue to update the Dynamo granule
-  try {
-    const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
-      knex,
-      deconstructCollectionId(granule.collectionId)
-    );
-
-    pgGranuleCumulusId = await granulePgModel.getRecordCumulusId(
-      knex,
-      {
-        granule_id: granule.granuleId,
-        collection_cumulus_id: collectionCumulusId,
-      }
-    );
-  } catch (error) {
-    if (!(error instanceof RecordDoesNotExist)) {
-      throw error;
-    }
-  }
-
+  const pgGranuleCumulusId = pgGranuleRecord.cumulus_id;
   let dynamoGranuleDeleted = false;
   try {
     return await knex.transaction(async (trx) => {
-      let pgGranule;
-      if (pgGranuleCumulusId) {
-        [pgGranule] = await granulePgModel.update(
-          trx,
-          {
-            cumulus_id: pgGranuleCumulusId,
-          },
-          {
-            published: false,
-            // using `undefined` would result in Knex ignoring this binding
-            // for the update. also, `undefined` is not a valid SQL value, it
-            // should be `null` instead
-            cmr_link: trx.raw('DEFAULT'),
-          },
-          ['*']
-        );
-      }
+      const [pgGranule] = await granulePgModel.update(
+        trx,
+        {
+          cumulus_id: pgGranuleCumulusId,
+        },
+        {
+          published: false,
+          // using `undefined` would result in Knex ignoring this binding
+          // for the update. also, `undefined` is not a valid SQL value, it
+          // should be `null` instead
+          cmr_link: trx.raw('DEFAULT'),
+        },
+        ['*']
+      );
       const dynamoGranule = await granuleDynamoModel.update(
-        { granuleId: granule.granuleId },
+        { granuleId: pgGranuleRecord.granule_id },
         { published: false },
         ['cmrLink']
       );
       dynamoGranuleDeleted = true;
-      await _removeGranuleFromCmr(granule);
+      const collectionId = await getGranuleCollectionId(knex, pgGranuleRecord);
```

Comment:
See comment on L4 - I think we might already have this via the apiGranule translation.

Reply:
Yeah, I think I'm not following the comment on Line 4. The record passed in is a PG Granule and I think we want that to drive this logic? I don't think there's an PG > API Granule translation in there...

We could get the collectionId directly from the `dynamoGranule` above but I think that sorta defeats the point. We'll be removing that anyway so including a lookup here based on the PG record seems to make sense but, again, I think I'm missing what you're saying.